### PR TITLE
[ContentFragmentList] Add UI Tests

### PR DESCRIPTION
**Pull request by jckautzmann**
Add UI tests (Hobbes) for the content fragment list component.
GH issue: https://github.com/adobe/aem-core-wcm-components/issues/522

### CODE REVIEW

File: testing/it/ui-js/src/content/jcr_root/conf/core-components/settings/dam/cfm/models/core-component-model/.content.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        <_x0031_547563167191
+                            jcr:primaryType="nt:unstructured"
+                            sling:resourceType="cq/gui/components/coral/common/form/tagfield"
+                            fieldLabel="Type"
+                            listOrder="7"
+                            metaType="tags"
+                            multiple="true"
+                            name="component-type"
+                            renderReadOnly="false"
+                            rootPath="/content/cq:tags/core-components-examples/component-type"
```
Comment (richardhand) [inline testing/it/ui-js/src/content/jcr_root/conf/core-components/settings/dam/cfm/models/core-component-model/.content.xml]: we shouldn't reuse library sample content in our test cases, such as these tags

File: testing/it/ui-js/src/content/jcr_root/content/dam/core-components/contentfragments-tests/image-fragment/.content.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                component-description="&lt;p>Image allows a single image asset to be displayed on the page. Images are adaptive; the relevant image width is selected for the screen size and lazy loading is available.&lt;/p>&#xa;"
+                component-description_x0040_ContentType="text/html"
+                component-description_x0040_LastModified="{Date}2019-03-29T19:09:36.629+01:00"
+                component-latest-version="{Long}2"
+                component-latest-version_x0040_LastModified="{Date}2019-03-29T19:09:36.629+01:00"
+                component-title="Image"
+                component-title_x0040_LastModified="{Date}2019-03-29T19:09:36.628+01:00"
+                component-type="[core-components-examples:component-type/basic]"
+                component-type_x0040_LastModified="{Date}2019-03-29T19:09:36.630+01:00"
+                type="[core-components-examples:component-type/basic]"
```
Comment (richardhand) [inline testing/it/ui-js/src/content/jcr_root/content/dam/core-components/contentfragments-tests/image-fragment/.content.xml]: same comment re. no library references in the test sample content

File: testing/it/ui-js/src/content/jcr_root/content/dam/core-components/contentfragments-tests/_jcr_content/folderThumbnail.dir/.content.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:dam="http://www.day.com/dam/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0" xmlns:nt="http://www.jcp.org/jcr/nt/1.0"
+    jcr:primaryType="nt:file">
+    <jcr:content
+        dam:folderThumbnailPaths="[/content/dam/core-components-examples/library/sample-assets/snowy-mountain-glacier.jpg,/content/dam/core-components-examples/library/sample-assets/lava-into-ocean.jpg,/content/dam/core-components-examples/library/sample-assets/lava-rock-formation.jpg]"
```
Comment (richardhand) [inline testing/it/ui-js/src/content/jcr_root/content/dam/core-components/contentfragments-tests/_jcr_content/folderThumbnail.dir/.content.xml]: I don't think we need to add this folderThumbnail at all for the test content, plus the references are from the library

File: testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/ContentFragmentList/v1/ContentFragmentList.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            .wait(200)
+            // select the title element
+            .execTestCase(c.tcUseDialogSelect("./elementNames", "component-title"))
+            // add a second element
+            .click(selectors.editDialog.elements.addButton)
+            .wait(200)
+            // expand the dropdown
+            .click(selectors.editDialog.elements.last + " " + selectors.editDialog.elements.select.button)
+            // select the type element
+            .click(selectors.editDialog.elements.last + " " + selectors.editDialog.elements.select.item + "[value='component-type']")
```
Comment (richardhand) [inline testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/ContentFragmentList/v1/ContentFragmentList.js]: we could add these element name identifiers to a variable for reuse:

```js
elementNames = {
    TYPE: 'component-type'
    TITLE: 'component-title'
}
```

```js
.click(selectors.editDialog.elements.last + " " + selectors.editDialog.elements.select.item + "[value='" + elementNames.TYPE + "']")
```

**Comment by jckautzmann** [comment]
Thx @richardhand for your feedback. I integrated it in commit https://github.com/adobe/aem-core-wcm-components/pull/547/commits/c1319f7b4c4e517910d83fca48497c7e2d4b8bc6

**Pull request merged**